modules/test/playwright/tests/commerce/commerceDiagramProduct.spec.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {mergeTests} from '@playwright/test';
+
+import {accountsPagesTest} from '../../fixtures/accountsPagesTest';
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {commercePagesTest} from '../../fixtures/commercePagesTest';
+import {loginTest} from '../../fixtures/loginTest';
+
+export const test = mergeTests(
+	apiHelpersTest,
+	accountsPagesTest,
+	commercePagesTest,
+	loginTest()
+);
+
+test('COMMERCE-11835 Account Supplier role user can upload diagram file/image', async ({
+	accountDetailsPage,
+	accountsPage,
+	apiHelpers,
+	commerceAdminProductDetailsDiagramPage,
+	commerceAdminProductDetailsPage,
+	commerceAdminProductPage,
+	page,
+}) => {
+	await page.goto('/');
+
+	const account = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'Supplier account',
+		type: 'supplier',
+	});
+
+	const catalog = await apiHelpers.headlessCommerceAdminCatalog.postCatalog({
+		accountId: account.id,
+	});
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account.id,
+		['test@liferay.com']
+	);
+
+	const product = await apiHelpers.headlessCommerceAdminCatalog.postProduct({
+		catalogId: catalog.id,
+		name: {
+			en_US: 'Product',
+		},
+		productType: 'diagram',
+	});
+
+	try {
+		await accountsPage.gotoAccount(account.name);
+
+		await accountDetailsPage.goToUsersTab();
+
+		await accountDetailsPage.goToAssignRoles('Account Supplier');
+
+		await commerceAdminProductPage.gotoProduct(product.name['en_US']);
+
+		await commerceAdminProductDetailsPage.goToProductDiagram();
+
+		await commerceAdminProductDetailsDiagramPage.goToDragAndDropImages();
```

Comment:
I'm not sure about this test because we don't have any expect in the test itself. Is is possible to move the 
```
await this.dragAndDropImages.waitFor({
			state: 'visible',
		});
```
as expect here?